node/delivery/deliver_client.go

```diff
@@ -63,18 +63,23 @@ func SingleSpecifiedSeekInfo(seq uint64) *orderer.SeekInfo {
 
 func Pull(context context.Context, channel string, logger types.Logger, endpoint func() string, requestEnvelopeFactory func() *common.Envelope, cc comm.ClientConfig, handleBlock func(block *common.Block), onClose func()) {
 	logger.Infof("Started pulling from: %s", channel)
-	for {
-		time.Sleep(time.Second)
 
-		select {
-		case <-context.Done():
-			logger.Infof("Returning since context is done")
-			if onClose != nil {
-				onClose()
-			}
-			return
-		default:
+	time.Sleep(time.Second)
+
+	select {
+	case <-context.Done():
+		logger.Infof("Returning since context is done")
+		if onClose != nil {
+			onClose()
 		}
+		return
+	default:
+	}
+
+	count := 0
+	retryInterval := minRetryInterval
+
+	for {
```

Comment:
if the endpoint() function return "" is there any hope of getting an actual endpoint? I think not... and then we are in a busy loop running full speed. Let's get rid of that... I don't think getting "" is valid for `endpoint()`